File: MandelbrotSet (Java)/src/ComplexNumber.java
//import com.sun.javafx.binding.StringFormatter;

import java.util.Formatter;

public class ComplexNumber {
    private double r, i;

    public ComplexNumber() {
        r = 0;
        i = 0;
    }

    public ComplexNumber(double _r, double _i) {
        r = _r;
        i = _i;
    }

    public ComplexNumber(ComplexNumber rhs) {
        this.r = rhs.r;
        this.i = rhs.i;
    }

    public void finalize() throws Throwable{
        super.finalize();
    }

    public void setR(double r) {
        this.r = r;
    }

    public void setI(double i) {
        this.i = i;
    }

    public double getR() {
        return r;
    }

    public double getI() {
        return i;
    }

    public String toString() {
        Formatter f = new Formatter();
        f.format("%.2f%s%.2f%s", r, (i >= +0 ? "+" : ""), i, "i");
        return f.toString();
    }

    ComplexNumber add(ComplexNumber rhs) {
        ComplexNumber cn = new ComplexNumber(this);
        cn.r += rhs.r;
        cn.i += rhs.i;
        return cn;
    }

    ComplexNumber sub(ComplexNumber rhs) {
        ComplexNumber cn = new ComplexNumber(this);
        cn.r -= rhs.r;
        cn.i -= rhs.i;
        return cn;
    }

    ComplexNumber mult(ComplexNumber rhs) {
        ComplexNumber cn = new ComplexNumber();
        cn.r = rhs.r * this.r - rhs.i * this.i;
        cn.i = rhs.r * this.i + rhs.i * this.r;
        return cn;
    }

    ComplexNumber div(ComplexNumber rhs) {
        ComplexNumber cn = new ComplexNumber();
        cn.r = (rhs.r * this.r + rhs.i * this.i) / (rhs.r * rhs.r + rhs.i * rhs.i);
        cn.i = (rhs.r * this.i - rhs.i * this.r) / (rhs.r * rhs.r + rhs.i * rhs.i);
        return cn;
    }

    double mag () {
        return Math.sqrt(r * r + i * i);
    }

    ComplexNumber conj () {
        ComplexNumber cn = new ComplexNumber(this);
        if(cn.i != 0)
            cn.i *= -1;
        return cn;
    }

    ComplexNumber sqrt () {
        ComplexNumber cn = new ComplexNumber();
        cn.r = Math.sqrt((r + Math.sqrt(r * r + i * i)) / 2);
        cn.i = Math.signum(i) * Math.sqrt((-r + Math.sqrt(r * r + i * i)) / 2);
        return cn;
    }

    boolean equals (ComplexNumber rhs) {
        if(rhs == null)
            return false;
        if(rhs.r != r)
            return false;
        if(rhs.i != i)
            return false;
        return true;
    }

}

File: MandelbrotSet C++/MandelbrotSet/ComplexNumber.cpp
//
//  ComplexNumber.cpp
//  ComplexNumber
//
//  Created by Raj Lakhanpal on 9/13/16.
//  Copyright © 2016 Raj Lakhanpal. All rights reserved.
//

#include "ComplexNumber.hpp"

//--constructors
ComplexNumber::ComplexNumber()
{
    re=0;
    im=0;
}
ComplexNumber::ComplexNumber(double r, double i) //:re(r), im(i)
{
    re = r;
    im = i;


}
ComplexNumber::ComplexNumber(const ComplexNumber &cn)
{
    re=cn.re;
    im=cn.im;
}

//--destructor
ComplexNumber::~ComplexNumber()
{

}

//--mutators
void ComplexNumber::setR(double r)
{
    re=r;
}
void ComplexNumber::setI(double i)
{
    im=i;
}

//--accessors
double ComplexNumber::getR() const
{
    return re;
}
double ComplexNumber::getI() const
{
    return im;
}

//--message handlers
void ComplexNumber::print() const
{
    std::cout << std::fixed << std::setprecision(1) << this->re << " + " << this->im << "i" << std::endl;
}

//--add two complex numbers
ComplexNumber ComplexNumber::add(const ComplexNumber &rhs) const
{
    return ComplexNumber(this-> re + rhs.re, this-> im + rhs.im);
}


//--subtract two complex numbers
ComplexNumber ComplexNumber::sub(const ComplexNumber &rhs) const
{
    return ComplexNumber(this-> re - rhs.re, this->im - rhs.im);
}

//--multiply two complex numbers
ComplexNumber ComplexNumber::mult(const ComplexNumber &rhs) const
{

    //return ComplexNumber(this-> re * rhs.re - im * rhs.im, this->im * rhs.im - re + rhs.re);
	return ComplexNumber(this->re * rhs.re - im * rhs.im, this->re * rhs.im + rhs.re * this->im);
}


//--divide two complex numbers
ComplexNumber ComplexNumber::div(const ComplexNumber &rhs) const
{

    return ComplexNumber(((this-> re) * (rhs.re) + ((im) * (rhs.im))) / (pow(rhs.re, 2) + pow(rhs.im, 2)), (((this-> im) * (rhs.re) - ((re) * (rhs.im))) / (pow(rhs.re, 2) + pow(rhs.im, 2))));
}


//--conjugate two complex numbers
ComplexNumber ComplexNumber::conj() const
{

    return ComplexNumber(this-> re, -1 * this->im);

}

double ComplexNumber::mag () const {
    return ::sqrt(this->re * this->re + this->im * this->im);
}




